File: week6/old-examples/csharp/src/OopOrganization.Chapter7/Domain/Geometry/Shapes/Circle.cs
using OopOrganization.Domain.Geometry.Contracts;

namespace OopOrganization.Domain.Geometry.Shapes;

public sealed class Circle : IShape
{
    public double Radius { get; }

    public Circle(double radius) => Radius = radius;

    public double Area() => Math.PI * Radius * Radius;

    public double Perimeter() => 2 * Math.PI * Radius;
}

File: week6/old-examples/javascript/src/domain/geometry/shapes/Circle.js
import { Shape } from "../contracts/Shape.js";

export class Circle extends Shape {
  constructor(radius) {
    super();
    this.radius = radius;
  }
  area() { return Math.PI * (this.radius ** 2); }
  perimeter() { return 2 * Math.PI * this.radius; }
}
